resource/airflow_dag_pipeline.yaml

```diff
@@ -49,6 +49,218 @@ jobs:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: concourse/docker-image-resource
+        inputs:
+          - name: pull-request
+        run:
+          path: sh
+          args:
+            - -ecx
+            - |
+              cp -R ./pull-request/* ./pull-request-edit
+              jq '.role_name = "test_dag_role"' ./pull-request/deploy.json > ./pull-request-edit/deploy.json
+              cat ./pull-request-edit/deploy.json
+        outputs:
+          - name: pull-request-edit
+
+  # test creating the IAM role
+  - aggregate:
+    - task: create-iam-role-name
+      file: common-tasks/put-iam-role.yaml
+      output_mapping: {output: aws}
+      input_mapping:
+        source: pull-request-edit
+      params:
+        ORG_NAME: ((github-org))
+        REPO_NAME: ((github-repo))
+        AWS_ACCESS_KEY_ID: ((secrets.role-putter-access-key-id))
+        AWS_SECRET_ACCESS_KEY: ((secrets.role-putter-secret-access-key))
+        AWS_NODES_ROLE_ARN: ((secrets.nodes-role-arn))
+        APP_NAME: ((app-name))
+      on_failure:
+        put: pull-request
+        attempts: 3
+        params:
+          path: pull-request
+          status: failure
+          context: iam-role
+    - task: run-test
+      privileged: true
+      config:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: amidos/dcind
+        inputs:
+          - name: image
+        run:
+          path: sh
+          args:
+            - -ec
+            - |
+              source /docker-lib.sh
+              start_docker
+              docker load -i image/image.tar
+              echo "Installing pytest"
+              docker run --entrypoint "" dag/test:latest pip install pytest
+              if [ -d "tests" ]; then
+                echo "Running tests in tests dir"
+                docker run --entrypoint "" dag/test:latest pytest tests
+              fi
+      on_failure:
+        put: pull-request
+        attempts: 3
+        params:
+          path: pull-request
+          status: failure
+          context: unittests
+    - task: run-syntax
+      privileged: true
+      config:
+        platform: linux
+        image_resource:
+          type: docker-image
+          source:
+            repository: amidos/dcind
+        inputs:
+          - name: image
+        run:
+          path: sh
+          args:
+            - -ec
+            - |
+              source /docker-lib.sh
+              start_docker
+              docker load -i image/image.tar
+              # Try importing files in base dir to check for syntax errors
+              docker run --entrypoint "" dag/test:latest python -c "
+              import os
+              from importlib import import_module
+              prefix = os.getcwd()
+              print(f'Working directory: {prefix}')
+              for dirpath, dirnames, filenames in os.walk(prefix):
```

Comment:
won't imports fail because airflow deps won't be installed?

Reply:
This runs in the actual job docker image. It just imports all the python files as a test for syntax errors. It works but might not be necessary.